source/MaterialXGenShader/ColorManagementSystem.cpp

```diff
@@ -71,11 +71,11 @@ ShaderNodePtr ColorManagementSystem::createNode(const ShaderGraph* parent, const
 
     // Create ports on the node.
     ShaderInput* input = shaderNode->addInput("in", transform.type);
-    if (transform.type == Type::COLOR3)
+    if (transform.type->getName() == Type::COLOR3->getName())
     {
         input->setValue(Value::createValue(Color3(0.0f, 0.0f, 0.0f)));
     }
-    else if (transform.type == Type::COLOR4)
+    else if (transform.type->getName() == Type::COLOR4->getName())
```

Comment:
Curios why this change was done, here and above, as it's now comparing strings instead of just a pointer compare.

Reply:
I'm not sure why but this pointer compare seems to be "unstable" when running through Python tests as it get's missed. When a string compare it seems to be stable. Could be the value set is not exactly these pointers so it fails. The error is that "color3" is not valid in the exception thrown.

Reply:
If that's the case it's a serious problem, as our codegen type system relies on these pointer comparison. It is what we use everywhere else for type checking, so it's surprising that it would fail in this case. So I think it's something we need to investigate further why it would fail.

Reply:
I have logged issue #1221 for this.